src/test/java/net/sf/jabref/logic/util/version/VersionTest.java

```diff
@@ -85,13 +106,42 @@ public void versionThreeDigitDevVersion() {
     }
 
     @Test
+    public void validVersionIsNotNewerThanUnknownVersion() {
+        // Reason: unknown version should only happen for developer builds where we don't want an update notification
+        Version unknownVersion = new Version(BuildInfo.UNKNOWN_VERSION);
+        Version validVersion = new Version("4.2");
+        assertFalse(validVersion.isNewerThan(unknownVersion));
+    }
+
+    @Test
     public void versionNewerThan() {
         Version olderVersion = new Version("2.4");
         Version newerVersion = new Version("4.2");
         assertTrue(newerVersion.isNewerThan(olderVersion));
     }
 
     @Test
+    public void versionNotNewerThanSameVersion() {
+        Version version1 = new Version("4.2");
+        Version version2 = new Version("4.2");
+        assertFalse(version1.isNewerThan(version2));
+    }
+
+    @Test
+    public void concreteVersionNewerThanUnknownVersion() {
+        Version concreteVersion = new Version("4.2");
+        Version unknownVersion = new Version(BuildInfo.UNKNOWN_VERSION);
+        assertTrue(concreteVersion.isNewerThan(unknownVersion));
```

Comment:
@koppor duplicates test above but with opposite assertion. As outlined above, I think it is better to have unknown > valid versions since otherwise we developers constantly get "new update" notifications on startup.